src/pyscaffold/templates/__init__.template

```diff
@@ -1,10 +1,15 @@
-from pkg_resources import DistributionNotFound, get_distribution
+import sys
+
+if sys.version_info[:2] >= (3, 8):
+    from importlib.metadata import PackageNotFoundError, version
+else:
+    from importlib_metadata import PackageNotFoundError, version
```

Comment:
This is the most controversial change in the entire PR, since it assumes we add `importlib_metadata` as a conditional runtime dependency (`install_requires`) in the generated package (which I include in the template and `update.py`)...

I feel like this would be more or less what the majority of people would do, and simply ditch `pkg_resources` completely (which is more future proof). Would it be OK @FlorianWilhelm?

The alternative is to keep using `pkg_resources` for Python < 3.8. Something like this:

```python
if sys.version_info[:2] >= (3, 8):
    from importlib.metadata import PackageNotFoundError, distribution
else:
    from pkg_resources import DistributionNotFound as PackageNotFoundError
    from pkg_resources import get_distribution as distribution

try:
    # Change here if project is renamed and does not equal the package name
    dist_name = ${distribution}
    __version__ = distribution(dist_name).version
except PackageNotFoundError:
    __version__ = "unknown"
finally:
    del distribution, PackageNotFoundError
```

Reply:
I like your PR solution. `importlib_metadata` is a backport of what's already established in 3.8 so you are exactly doing what the Python devs want. We could maybe add a `ToDo` comment stating that this can be removed when we have `python_requires = >=3.8` one day.